features/timeseries/src/main/java/org/opennms/netmgt/timeseries/integration/persistence/TimeseriesPersistOperationBuilder.java

```diff
@@ -122,8 +122,10 @@ public void setAttributeMetadata(String metricIdentifier, String name) {
 
     @Override
     public void commit() {
-        writer.insert(getSamplesToInsert());
-        writer.index(getSamplesToIndex());
+        try(final Timer.Context context = commitTimer.time()) {
+            writer.insert(getSamplesToInsert());
+            writer.index(getSamplesToIndex());
+        }
```

Comment:
This metric may be not detailed enough. I see two problems here:
* This measures to calls to the implementation which we maybe want to inspect separately.
* We also measure the time needed to build the list of samples to insert/index.

Reply:
> This measures to calls to the implementation which we maybe want to inspect separately.

True, we measure the overall time to collect the samples. You think we should turn that into 2 separate timers?

> We also measure the time needed to build the list of samples to insert/index.

The actual insert happens in a different thread (managed by the ringbuffer).
We measure the insert times separately:
metadata.write.db
samples.write.ts

Reply:
Ok, if this is the collect part only, a single metric is fine